src/content/guides/progressive-web-application.md

```diff
@@ -0,0 +1,150 @@
+
+This section will focus on adding an offline experience to our app. We'll achieve this using a Google project called [Workbox](https://github.com/GoogleChrome/workbox) which provides tools that help make offline support for web apps easier to setup.
+
+## We Don't Work Offline Now
+
+So far, when we've been browsing our webpack output, we've been going directly to the local file system. That's not the typical experience a user has when accessing a web app. Normally a user accesses a web app over a network; their browser talking to a __server__ which will serve up the required assets; `html`, `js`, `css` etc.
+
+So let's test what the current experience is like using a simple server. Let's use the [http-server](https://www.npmjs.com/package/http-server) package: `npm install http-server --save-dev`. We'll also amend the `scripts` section of our `package.json` to add in a `start` script:
+
+``` diff
+{
+  ...
+  "scripts": {
+-    "build": "webpack"
++    "build": "webpack",
++    "start": "http-server dist"
+  },
+  ...
+}
+```
+
+If you haven't previously done so, run the command `npm run build` to build your project. Then run the command `npm run start`. This should produce output like this:
+
+``` bash
+> http-server dist
+
+Starting up http-server, serving dist
+Available on:
+  http://10.0.75.1:8080
+  http://127.0.0.1:8080
+  http://192.168.0.7:8080
+Hit CTRL-C to stop the server
+```
+
+If you open your browser to http://localhost:8080 you should see your webpack application being served up from the `dist` directory. If you stop the server and refresh, the webpack application is no longer available.  
+
+This is what we aim to change. Once we reach the end of this module we should be able to stop the server, hit refresh and still see our application.
+
+## Adding Workbox
+
+Let's add the Workbox webpack plugin and adjust the `webpack.config.js` file:
+
+``` bash
+npm install workbox-webpack-plugin --save-dev
+```
+
+__webpack.config.js__
+
+``` diff
+  const path = require('path');
+  const HtmlWebpackPlugin = require('html-webpack-plugin');
+  const CleanWebpackPlugin = require('clean-webpack-plugin');
++ const WorkboxPlugin = require('workbox-webpack-plugin');
+
+  module.exports = {
+    entry: {
+      app: './src/index.js',
+      print: './src/print.js'
+    },
+    plugins: [
+        new CleanWebpackPlugin(['dist']),
+        new HtmlWebpackPlugin({
+            title: 'Output Management'
+-       })
++       }),
++       new WorkboxPlugin({
++           // these options encourage the ServiceWorkers to get in there fast 
++           // and not allow any straggling "old" SWs to hang around
++           clientsClaim: true,
++           skipWaiting: true,
```

Comment:
Can you remove the trailing comma? I know it's valid ES6 now but we've consistently left them out in the docs.

Reply:
done